presto-main/src/main/java/com/facebook/presto/statistics/PlanNodeStatistics.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statistics;
+
+import com.facebook.presto.spi.statistics.ValueEstimate;
+
+import java.util.function.Function;
+
+import static com.facebook.presto.spi.statistics.ValueEstimate.unknownValue;
+import static java.util.Objects.requireNonNull;
+
+public class PlanNodeStatistics
+{
+    public static final PlanNodeStatistics EMPTY_STATISTICS = PlanNodeStatistics.builder().build();
+
+    private final ValueEstimate outputRowsCount;
+    private final ValueEstimate outputSizeInBytes;
+
+    private PlanNodeStatistics(ValueEstimate outputRowsCount, ValueEstimate outputSizeInBytes)
+    {
+        this.outputRowsCount = requireNonNull(outputRowsCount, "outputRowsCount can not be null");
+        this.outputSizeInBytes = requireNonNull(outputSizeInBytes, "outputSizeInBytes can not be null");
+    }
+
+    public ValueEstimate getOutputRowsCount()
+    {
+        return outputRowsCount;
+    }
+
+    public ValueEstimate getOutputSizeInBytes()
+    {
+        return outputSizeInBytes;
+    }
+
+    public PlanNodeStatistics mapOutputRowsCount(Function<Double, Double> mappingFunction)
+    {
+        return builder().setFrom(this).setOutputRowsCount(outputRowsCount.map(mappingFunction)).build();
+    }
+
+    public PlanNodeStatistics mapOutputSizeInBytes(Function<Double, Double> mappingFunction)
+    {
+        return builder().setFrom(this).setOutputSizeInBytes(outputRowsCount.map(mappingFunction)).build();
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
```

Comment:
I have recently below way for Builder pattern to be more concise. You may want to consider this
```
 public static class NiceObject
    {
        private final Value value;

        public static Builder builder()
        {
            return new NiceObject().new Builder();
        }

        private NiceObject()
        {
        }
 
...
      
        public class Builder
        {
            private boolean builidng = true; // optional for nazi-safers
            public Builder set(Value value)
            {
               checkState(building);
               this.value = value;
               return this;
            }

            public TeamRaterStats build()
            {
                buidling = false;
                return NiceObject.this;
            }
        }
    }
```